Documentation/Registry.md

```diff
@@ -661,31 +673,45 @@ as described by [RFC 6249].
 A client MAY use this information
 to determine its preferred strategy for downloading.
 
-A server that indexes but doesn't host packages
-MAY respond with a status code of `303` (See Other)
-and redirect to a hosted package archive if one is available.
+A server MAY respond with a status code of `303` (See Other)
+to redirect the client to download the source archive from another host.
+The client MUST NOT follow redirects that downgrade to an insecure connection.
+The client SHOULD limit the number of redirects to prevent a redirect loop.
+
+For example,
+a server redirects the client to download from
+a content delivery network (CDN) using a signed URL:
 
 ```http
 HTTP/1.1 303 See Other
-Content-Version: 1
+Location: https://example.cdn.com/LinkedList-1.1.1.zip?key=XXXXXXXXXXXXXXXXX
+```
+
+```http
+GET /LinkedList-1.1.1.zip?key=XXXXXXXXXXXXXXXXX HTTP/1.1
+Host: example.cdn.com
+Accept: application/vnd.swift.registry.v1+zip
+```
+
+```http
+HTTP/1.1 200 OK
+Accept-Ranges: bytes
+Cache-Control: public, immutable
+Content-Type: application/zip
+Content-Disposition: attachment; filename="LinkedList-1.1.1.zip"
 Content-Length: 2048
+Content-Version: 1
 Digest: sha-256=a2ac54cf25fbc1ad0028f03f0aa4b96833b83bb05a14e510892bb27dea4dc812
-Location: https://packages.example.com/mona/LinkedList/1.1.1.zip
 ```
 
-The client SHOULD consider the `Digest` and `Content-Length` headers
-sent in the response to be authoritative for the redirected resource.
-The client MUST NOT follow redirects that downgrade to an insecure connection.
-The client SHOULD limit the number of redirects to prevent a redirect loop.
-
 <a name="endpoint-5"></a>
 
 ### 4.5 Lookup package identifiers registered for a URL
 
 A client MAY send a `GET` request
 for a URI matching the expression `/identifiers{?url}`
 to retrieve package identifiers associated with a particular URL.
-A client SHOULD set the `Accept` header to
+A client SHOULD set the `Accept` header with the value
 `application/vnd.swift.registry.v1+json`.
 
 ```http
```

Comment:
The response for this API is an array of identifier strings:

```
{
    "identifiers": [
      "mona.LinkedList"
    ]
}
```

Would it better if it were an array of objects?

```
{
  "identifiers": [
    { 
      "scope": "mona",
      "name": "LinkedList"
    }
  ]
}
```

Reply:
The purpose of this endpoint is to give the registry a way to map existing URL-based dependencies with identifiers so that SwiftPM can resolve a mixed dependency graph.

In the case of registry configuration (`.swiftpm/config/registries.json`), it made sense to have an object (`"[default]": {"url": "https://..."}`) instead of a single value (`"[default]": "https://"`) because that would provide an extension point if we needed additional information later, like for authentication.

Here, I can't think of anything else we'd need to convey beyond the identifier. Decomposing the identifier into fields doesn't strike me as particularly necessary. Can you think of anything else we'd want to put here?

Reply:
I am trying to consider how the received data will be used. It's fine to keep identifiers as strings if we are just going to use them as-is, but if we ever need to extract scope and/or name then it might be more convenient to represent them as objects instead.